File: DOTNET BACKEND/TransitHive/Models/Booking.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using Stripe;

namespace TransitHive.Models
{
    public class Booking
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Mobile { get; set; }
        public string Email { get; set; }
        public string City { get; set; }
        public DateTime MoveDate { get; set; }
        public string PickupLocation { get; set; }
        public string DropLocation { get; set; }

        [Column(TypeName = "ENUM('PENDING', 'APPROVED', 'SUSPENDED')")]
        public LiftOption PickupLift { get; set; }
        public int? PickupFloors { get; set; }

        [Column(TypeName = "ENUM('YES','NO')")]
        public LiftOption DropLift { get; set; }
        public int? DropFloors { get; set; }
        public String PackagingType { get; set; }
        public int? VendorId { get; set; }
        public int UserId { get; set; }


        [Column(TypeName = "ENUM('PENDING', 'CONFIRMED', 'ASSIGNED', 'COMPLETED', 'CANCELLED')")]
        public BookingStatus Status { get; set; } = BookingStatus.PENDING;
        public decimal Cost { get; set; }

        public bool IsPaid { get; set; } = false;
        public bool IsReviewed { get; set; } = false;

        [Column(TypeName = "ENUM('CASH','CARD')")]
        public PaymentM PaymentMethod { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.Now;

        // Navigation properties
        public User User { get; set; }
        public Vendor Vendor { get; set; }
        public ICollection<BookingItem> Items { get; set; }
        public ICollection<Payment> Payments { get; set; }
        public ICollection<Review> Reviews { get; set; }
    }

    public enum LiftOption { YES, NO }
    //public enum PackagingType { SingleLayered, DoubleLayered, Wooden }
    public enum BookingStatus { PENDING, CONFIRMED, ASSIGNED, COMPLETED, CANCELLED }

    public enum PaymentM { CASH, CARD }
}

File: SpringBootBackend/src/main/java/com/th/entity/Booking.java
package com.th.entity;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
public class Booking extends BaseEntity {
    private String name;
    private String mobile;
    private String email;
    private String city;
    private LocalDate moveDate;
    private String pickupLocation;
    private String dropLocation;
    private Integer pickupFloors;
    private Integer dropFloors;
    private String packagingType;
    private BigDecimal cost;
    
    @Enumerated(EnumType.STRING)
    private PaymentM paymentMethod;
    
    private boolean isPaid;
    
    private boolean isReviewed;    
    

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "vendor_id", nullable = true)
    private Vendor vendor;

    @OneToMany(mappedBy = "booking", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<BookingItem> bookingItems = new ArrayList<>();

    @Enumerated(EnumType.STRING)
    private Status status = Status.PENDING;

    @Enumerated(EnumType.STRING)
    private LiftAvailability pickupLift;

    @Enumerated(EnumType.STRING)
    private LiftAvailability dropLift;

    public enum Status {
        PENDING, CONFIRMED, ASSIGNED, COMPLETED, CANCELLED
    }

    public enum LiftAvailability {
        YES, NO
    }
    
    public enum PaymentM { CASH, CARD }
}
